client/asset/eth/eth.go

```diff
@@ -1000,8 +966,29 @@ func (eth *ExchangeWallet) findSecret(ctx context.Context, secretHash [32]byte,
 
 // Refund refunds a contract. This can only be used after the time lock has
 // expired.
-func (*ExchangeWallet) Refund(coinID, contract dex.Bytes, feeSuggestion uint64) (dex.Bytes, error) {
-	return nil, asset.ErrNotImplemented
+func (eth *ExchangeWallet) Refund(_, contract dex.Bytes, feeSuggestion uint64) (dex.Bytes, error) {
+	version, secretHash, err := dexeth.DecodeContractData(contract)
+	if err != nil {
+		return nil, fmt.Errorf("Refund: failed to decode contract: %w", err)
```

Comment:
Just a silly suggestion, so feel free to ignore. If you would like to put the method name in every error message, you could define it at the beginning like `const op = "Refund"` then do `fmt.Errorf("%s: failed: %v", op, err)`

Really just a thing I have seen ppl do if you like it. Probably prevents some typos and repitition? If the method name were to change can get all the messages at once. No need to change at all though, just a suggestion.